modules/custom/az_core/src/AZCoreGuzzleCache.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace Drupal\az_core;
+
+use Drupal\az_core\Strategy\AZPoliteCacheStrategy;
+use Drupal\Core\Cache\BackendChain;
+use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\Core\Cache\MemoryBackend;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Kevinrob\GuzzleCache\CacheEntry;
+use Kevinrob\GuzzleCache\CacheMiddleware;
+use Kevinrob\GuzzleCache\Storage\CacheStorageInterface;
+use Kevinrob\GuzzleCache\Strategy\NullCacheStrategy;
+
+/**
+ * Provides a Drupal cache backend for the Guzzle caching middleware.
+ */
+class AZCoreGuzzleCache implements CacheStorageInterface {
```

Comment:
Nitpick: I think naming this class `AZCoreGuzzleCacheStorage` would be more consistent with the storage classes included with the `Kevinrob/guzzle-cache-middleware` (see https://github.com/Kevinrob/guzzle-cache-middleware/tree/master/src/Storage)
```suggestion
class AZCoreGuzzleCacheStorage implements CacheStorageInterface {
```

Reply:
But maybe this is doing more than just providing the storage?